packages/gatsby-source-filesystem/src/create-remote-file-node.js

```diff
@@ -100,15 +117,7 @@ module.exports = ({ url, store, cache, createNode, auth = {} }) => {
         fs.removeSync(tmpFilename)
       }
 
-      // Create the file node and return.
-      createFileNode(filename, {}).then(fileNode => {
-        // Override the default plugin as gatsby-source-filesystem needs to
-        // be the owner of File nodes or there'll be conflicts if any other
-        // File nodes are created through normal usages of
-        // gatsby-source-filesystem.
-        createNode(fileNode, { name: `gatsby-source-filesystem` })
-        resolve(fileNode)
-      })
+      createFileSystemNode()
```

Comment:
Could you not create a new function please

Reply:
I can change it, how would you prefer it to be handled?
The code needs to `createFileNode` whether or not it has to download the remote file, and originally that function was being called on `responseStream.on(`end)'.
Would you rather I just write out the function twice?

Reply:
Hmmm I'm not sure what this PR is fixing now. Files aren't redownloaded — a 304 response has an empty body.